pkg/controller/daemon/daemon_controller.go

```diff
@@ -911,7 +908,8 @@ func (dsc *DaemonSetsController) syncNodes(ds *extensions.DaemonSet, podsToDelet
 
 	glog.V(4).Infof("Nodes needing daemon pods for daemon set %s: %+v, creating %d", ds.Name, nodesNeedingDaemonPods, createDiff)
 	createWait := sync.WaitGroup{}
-	template := util.CreatePodTemplate(ds.Spec.Template, ds.Spec.TemplateGeneration, hash)
+	generation, _ := util.GetTemplateGeneration(ds)
```

Comment:
I think it's fine to ignore the returned error here. However, if there's an error, we should set `generation` to `nil` to avoid unexpected behavior.

Reply:
Do we need to print a warning here if there's an error?